File: hello-web/utils/health-check.ts
import * as http from 'http';

function checkHealth(url: string): void {
  http
    .get(url, (res: http.IncomingMessage) => {
      let data = '';

      res.on('data', (chunk: Buffer) => {
        data += chunk;
      });

      res.on('end', () => {
        if (res.statusCode === 200) {
          console.log(`HTTP Health Check for ${url} - OK`);
        } else {
          console.log(`HTTP Health Check for ${url} - Failed (Status Code: ${res.statusCode})`);
          process.exit(1);
        }
      });
    })
    .on('error', (error: Error) => {
      console.error('Error making request:', error.message);
      process.exit(1);
    });
}

const url: string = process.argv[2];

if (!url) {
  console.error('Usage: node health-check.js <URL>');
  process.exit(1);
}

checkHealth(url);

File: hello-user/utils/health_check.py
import urllib.request
import sys

def check_health(url):
    try:
        response = urllib.request.urlopen(url)
        if response.getcode() == 200:
            print(f"HTTP Health Check for {url} - OK")
        else:
            print(f"HTTP Health Check for {url} - Failed (Status Code: {response.getcode()})")
            sys.exit(1)
    except urllib.error.URLError as e:
        print(f"HTTP Health Check for {url} - Failed ({e})")
        sys.exit(1)

if __name__ == "__main__":
    if len(sys.argv) != 2:
        print("Usage: python health_check.py <URL>")
        sys.exit(1)
    url = sys.argv[1]
    check_health(url)
